SplashScreen/app/src/main/java/com/example/android/splashscreen/CustomActivity.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.splashscreen
+
+import android.animation.ObjectAnimator
+import android.os.Bundle
+import android.os.SystemClock
+import android.view.View
+import android.view.animation.AnticipateInterpolator
+import androidx.core.animation.doOnEnd
+
+/**
+ * "Custom Splash Screen". This is similar to [AnimatedActivity], but also has a custom animation
+ * for the splash screen exit.
+ */
+class CustomActivity : MainActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // This callback is called when the app is ready to draw its content and replace the splash
+        // screen. We can customize the exit animation of the splash screen here.
+        splashScreen.setOnExitAnimationListener { splashScreenView ->
+
+            // The animated vector drawable is already animating at this point. Depending on the
+            // duration of the app launch, the animation might not have finished yet. We can
+            // calculate the remaining duration of the icon animation as follows.
+            val remainingDuration = (
+                    splashScreenView.iconAnimationDurationMillis -
```

Comment:
This looks a bit weird to me. Can this be extracted to an ext-val?

Reply:
Extracted the calculation and added comments about each of the involved values.